File: snapshots/input/syntax/Main/Preprocessors.cs
namespace Main;

public class Preprocessors
{
    string OS()
    {
#if WIN32
        string os = "Win32";
#warning This class is bad.
#error Okay, just stop.
#elif MACOS
        string os = "MacOS";
#else
        string os = "Unknown";
#endif
        return os;
    }
}

File: snapshots/input/syntax/VBMain/Preprocessors.vb
Imports System.Diagnostics.CodeAnalysis

Namespace VBMain
    <SuppressMessage("ReSharper", "all")>
    Public Class Preprocessors
        Private Function OperatingSystem() As String
#If WIN32 Then
            Dim Os As String = "Win32"
#warning This class is bad.
#error Okay, just stop.
#elif MACOS
            Dim Os As String = "MacOS"
#Else
            Dim Os As String = "Unknown"
#End If
            Return Os
        End Function
    End Class
End Namespace